cvat-ui/src/components/annotation-page/tag-annotation-workspace/tag-annotation-sidebar/shortcuts-select.tsx

```diff
@@ -131,29 +70,34 @@ function ShortcutsSelect(props: Props): JSX.Element {
         setShortcutLabelMap(newShortcutLabelMap);
     }, []);
 
+    useDynamicLabels(componentShortcuts);
+
     useEffect(() => {
         const updatedComponentShortcuts = {
-            ...componentShortcuts,
+            ...Object.keys(componentShortcuts).reduce((acc: any, key) => {
+                if (keyMap[key]) {
```

Comment:
May you please explain when this condition may be falsy?

Reply:
As I understand it is only in order to TypeScript does not show any warnings
From the point of code, it would be clearer if we just have an action to update existing shortcuts.

This method would accept partial body for many shortcuts and apply it to the storage, e.g.

```
function updateShortcutAction({ shortcutID: Partial<ShortcutBody> }) ... 
```

Otherwise we need checks like this and a little weird thing about it is:
We refer to `keyMap` in `useEffect`, and according to idea, `keyMap` has to be in useEffect dependencies. However I understand it will lead to infinite rendering loop.

Reply:
By the way, probably the same issue forces you to use `useRef` in hook to reset defaults